crypto/s2n_fips.c

```diff
@@ -26,18 +26,25 @@
 
 static bool s2n_fips_mode_enabled = false;
 
-/* FIPS mode can be checked if OpenSSL was configured and built for FIPS which then defines OPENSSL_FIPS.
+/* FIPS mode can be checked if OpenSSL was configured and built for FIPS which
+ * then defines OPENSSL_FIPS.
  *
- * AWS-LC always defines FIPS_mode() that you can call and check what the library was built with. It does not define
- * a public OPENSSL_FIPS/AWSLC_FIPS macro that we can (or need to) check here
+ * AWS-LC always defines FIPS_mode() that you can call and check what the
+ * library was built with. It does not define a public OPENSSL_FIPS/AWSLC_FIPS
+ * macro that we can (or need to) check here
  *
- * Safeguard with macro's, for example because Libressl dosn't define
+ * Safeguard with macro's, for example because Libressl doesn't define
  * FIPS_mode() by default.
  *
- * Note: FIPS_mode() does not change the FIPS state of libcrypto. This only returns the current state. Applications
- * using s2n must call FIPS_mode_set(1) prior to s2n_init.
- * */
-bool s2n_libcrypto_is_fips(void)
+ * Note: FIPS_mode() does not change the FIPS state of libcrypto. This only
+ * returns the current state. Applications using s2n must call FIPS_mode_set(1)
+ * prior to s2n_init.
+ *
+ * Note: Developers should use `s2n_is_in_fips_mode()` instead of calling this
+ * directly. `s2n_is_in_fips_mode()` returns libcrypto FIPS status at library
+ * initialization, ie. s2n_init().
+ */
+bool s2n_query_libcrypto_is_fips(void)
```

Comment:
This change isn't actually related to the engine feature probe, and might be worth a separate PR.

I really dislike the new name. It didn't match the module name before, but now it even starts with a verb, which is unusual for an s2n method. "query" is also an unusual choice of wording that makes it sound like more than just a library call.

I don't hate the old name, but if you're trying to improve it maybe "s2n_fips_init_from_libcrypto" or "s2n_fips_get_libcrypto_status". Alternatively, just get rid of this method completely and move its logic into s2n_fips_init.

Reply:
I'll move this out to a separate PR.

In production, this fn should only be called once from init. Calling this fn directly could lead to errors since a libcryptos FIPS_mode() could have changed after the call to s2n_init().

The query name was to highlight that we are infact querying the libcrypto's FIPS_mode() status. Moving the logic to init would be ideal, except we do need to call this fn from a test that is testing s2n_init() behavior.